cmd/common.go

```diff
@@ -430,6 +385,102 @@ func Lookup(c *cliclient.MasterClient, name string, types ...string) (*ntypes.Re
 	return byName, nil
 }
 
+func lookupSchemaClient(c *cliclient.MasterClient, schemaType string) (clientbase.APIBaseClientInterface, error) {
+	rt, err := GetResourceType(c, schemaType)
+	if err != nil {
+		logrus.Debugf("Error GetResourceType: %v", err)
+		return nil, err
+	}
+	var schemaClient clientbase.APIBaseClientInterface
+	// the schemaType dictates which client we need to use
+	if c.CAPIClient != nil {
+		if strings.HasPrefix(rt, "cluster.x-k8s.io") {
+			schemaClient = c.CAPIClient
+		}
+	}
+	if c.ManagementClient != nil {
+		if _, ok := c.ManagementClient.APIBaseClient.Types[rt]; ok {
+			schemaClient = c.ManagementClient
+		}
+	}
+	if c.ProjectClient != nil {
+		if _, ok := c.ProjectClient.APIBaseClient.Types[rt]; ok {
+			schemaClient = c.ProjectClient
+		}
+	}
+	if c.ClusterClient != nil {
+		if _, ok := c.ClusterClient.APIBaseClient.Types[rt]; ok {
+			schemaClient = c.ClusterClient
+		}
+	}
+	return schemaClient, nil
+}
+
+func LookupByName(c *cliclient.MasterClient, name, schemaType string) (*ntypes.Resource, error) {
+	schemaClient, err := lookupSchemaClient(c, schemaType)
+	if err != nil {
+		return nil, err
+	}
+	var collection ntypes.ResourceCollection
+
+	listOpts := &ntypes.ListOpts{
+		Filters: map[string]interface{}{
+			"name":         name,
+			"removed_null": 1,
+		},
+	}
+
+	if err := schemaClient.List(schemaType, listOpts, &collection); err != nil {
+		logrus.Debugf("Error schemaClient.List: %v", err)
+		if clientbase.IsNotFound(err) {
+			return nil, fmt.Errorf("Not found: %s", name)
+		}
+		// prevents 403 from falsely returning as an unhandled error
+		if apiError, ok := err.(*clientbase.APIError); ok {
+			if apiError.StatusCode == http.StatusForbidden {
+				return nil, fmt.Errorf("Not found: %s", name)
+			}
+		}
+		return nil, err
+	}
+
+	if len(collection.Data) > 1 {
+		ids := []string{}
+		for _, data := range collection.Data {
+			ids = append(ids, data.ID)
+		}
+		return nil, fmt.Errorf("Multiple resources of type %s found for name %s: %v", schemaType, name, ids)
+	}
+	if len(collection.Data) == 0 {
```

Comment:
Should we return a "Not Found" error here as well?

Reply:
it seems like i had a reason not to, but can't see it now.. so I'll ad a not found return